src/Simulator/FloodyFluidSimulator.cpp

```diff
@@ -427,5 +429,39 @@ bool cFloodyFluidSimulator::HardenBlock(cChunk * a_Chunk, int a_RelX, int a_RelY
 	return false;
 }
 
+void cFloodyFluidSimulator::SoakNeighbors(cChunk * a_Chunk, Vector3i a_Rel, BLOCKTYPE a_BlockType, NIBBLETYPE a_Meta)
+{
+	// Only water blocks can soak neighboring blocks
+	if (!IsBlockWater(a_BlockType))
+	{
+		return;
+	}
+
+	BLOCKTYPE BlockType;
+	NIBBLETYPE BlockMeta;
+	// The blocks adjacent and below this water can be soaked
+	Vector3i Coords[] =
+	{
+		a_Rel + Vector3i( 1,  0,  0),
+		a_Rel + Vector3i(-1,  0,  0),
+		a_Rel + Vector3i( 0,  0,  1),
+		a_Rel + Vector3i( 0,  0, -1),
+		a_Rel + Vector3i( 0, -1,  0),
+	};
```

Comment:
Prefer a static array of offsets, then add them to `a_Rel` in the for-loop.

Reply:
Done